adapters/json_parse_test.go

```diff
@@ -94,3 +95,31 @@ func TestJsonParse_Perform(t *testing.T) {
 		})
 	}
 }
+
+func TestJSON_UnmarshalJSON(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name      string
+		input     string
+		want      []string
+		wantError bool
+	}{
+		{"array", `{"path":["1","b"]}`, []string{"1", "b"}, false},
+		{"array with dots", `{"path":["1",".","b"]}`, []string{"1", ".", "b"}, false},
+		{"string", `{"path":"first"}`, []string{"first"}, false},
+		{"dot delimited", `{"path":"1.b"}`, []string{"1", "b"}, false},
+		{"dot delimited empty string", `{"path":"1...b"}`, []string{"1", "", "", "b"}, false},
+		{"unclosed array errors", `{"path":["1"}`, []string{}, true},
+		{"unclosed string errors", `{"path":"1.2}`, []string{}, true},
```

Comment:
Related to the question above, the implementation supports unquoted strings, but does not test them, if we feel that unquoted input should be supported I feel like we should test that.

Reply:
I don't think it will support unquoted strings, but can write a test for that if needed.

Reply:
Yeah, unquoted strings error out.

Reply:
I don't believe JSON supports unquoted strings, so what would it be processing?

Reply:
Idk, maybe I misinterpreted what you thought it'd be doing. Sounds like we agree, no need to support unquoted input.